app/src/test/java/com/codesoom/assignment/application/CatToyServiceTestNested.java

```diff
@@ -0,0 +1,249 @@
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.ArgumentsProvider;
+import org.junit.jupiter.params.provider.ArgumentsSource;
+import org.springframework.test.util.ReflectionTestUtils;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+
+import static com.codesoom.assignment.Constant.IMAGE_URL;
+import static com.codesoom.assignment.Constant.MAKER;
+import static com.codesoom.assignment.Constant.NAME;
+import static com.codesoom.assignment.Constant.PRICE;
+import static com.codesoom.assignment.Constant.OTHER_IMAGE_URL;
+import static com.codesoom.assignment.Constant.OTHER_MAKER;
+import static com.codesoom.assignment.Constant.OTHER_NAME;
+import static com.codesoom.assignment.Constant.OTHER_PRICE;
+
+@DisplayName("CatToyService 서비스 테스트")
+public class CatToyServiceTestNested {
+
+    private CatToyService service;
+    private CatToyRepository repository;
+
+    private CatToy catToy;
+
+    @BeforeEach
+    void setUp() {
+
+        repository = mock(CatToyRepository.class);
+
+        service = new CatToyService(repository);
+
+        setUpFixtures();
+    }
+
+    private void setUpFixtures() {
+        catToy = new CatToy(1L, NAME, MAKER, PRICE, IMAGE_URL);
+
+        given(repository.findAll())
+                .willReturn(new ArrayList<>())
+                .willReturn(Collections.singletonList(catToy));
+
+        given(repository.findById(1L)).willReturn(Optional.of(catToy));
+
+        given(repository.findById(100L)).willThrow(new CatToyNotFoundException(100L));
+
+        given(repository.save(any(CatToy.class))).willReturn(catToy);
+
+    }
+
+    @Nested
+    @DisplayName("findAll 메서드는")
+    class Describe_findAll {
+        @Nested
+        @DisplayName("저장된 장난감들이 없을 경우")
+        class Context_without_toy {
+            @DisplayName("빈 목록이 반환된다.")
+            @Test
+            void findAllNotExistsToy() {
+                final List<CatToy> catToys = service.findAll();
+
+                assertThat(catToys).isEmpty();
+            }
+        }
+
+        @Nested
+        @DisplayName("저장된 장난감이 있을 경우")
+        class Context_with_toy {
+            @DisplayName("목록이 반환된다.")
+            @Test
+            void findAllExistsToy() {
+                assertThat(service.findAll()).isEmpty();
+
+                service.save(catToy);
+
+                assertThat(service.findAll()).isNotEmpty();
+            }
+        }
```

Comment:
테스트 준비는 `Context` 스코프에서 실행해주세요. 
```suggestion
        @Nested
        @DisplayName("저장된 장난감이 있을 경우")
        class Context_with_toy {
            @BeforeEach
            void prepareSavedToys() {
                service.save(catToy);
            }

            @DisplayName("목록이 반환된다.")
            @Test
            void findAllExistsToy() {
                assertThat(service.findAll()).isNotEmpty();
            }
        }
```

손인준님의 PR에 남긴 코멘트를 읽어 보시길 권해드려요.
https://github.com/CodeSoom/spring-week3-assignment-1/pull/35#discussion_r697871634